### ISSUE TITLE
Cloning master with no modifications results in an error

### ISSUE THREAD

**Issue by Skinner927** [mikecao/flight#73]
array_key_exists(): The first argument should be either a string or an integer (2)
#0 [internal function]: flight\Engine->handleError(2, 'array_key_exist...', 'C:\wamp\www\tim...', 87, Array)
#1 C:\wamp\www\timeclock\flight\net\Response.php(87): array_key_exists(Array, Array)
#2 C:\wamp\www\timeclock\flight\Engine.php(325): flight\net\Response->status(Array)
#3 C:\wamp\www\timeclock\flight\core\Dispatcher.php(198): flight\Engine->_stop(Array, NULL)
#4 C:\wamp\www\timeclock\flight\core\Dispatcher.php(141): flight\core\Dispatcher::invokeMethod(Array, Array)
#5 C:\wamp\www\timeclock\flight\core\Dispatcher.php(48): flight\core\Dispatcher::execute(Array, Array)
#6 C:\wamp\www\timeclock\flight\Engine.php(63): flight\core\Dispatcher->run('stop', Array)
#7 C:\wamp\www\timeclock\flight\core\Dispatcher.php(198): flight\Engine->__call('stop', Array)
#8 C:\wamp\www\timeclock\flight\core\Dispatcher.php(198): flight\Engine->stop(Array, NULL)
#9 C:\wamp\www\timeclock\flight\core\Dispatcher.php(141): flight\core\Dispatcher::invokeMethod(Array, Array)
#10 C:\wamp\www\timeclock\flight\core\Dispatcher.php(126): flight\core\Dispatcher::execute(Array, Array)
#11 C:\wamp\www\timeclock\flight\core\Dispatcher.php(52): flight\core\Dispatcher->filter(Array, Array, NULL)
#12 C:\wamp\www\timeclock\flight\Engine.php(63): flight\core\Dispatcher->run('start', Array)
#13 C:\wamp\www\timeclock\flight\core\Dispatcher.php(190): flight\Engine->__call('start', Array)
#14 C:\wamp\www\timeclock\flight\core\Dispatcher.php(190): flight\Engine->start()
#15 C:\wamp\www\timeclock\flight\Flight.php(43): flight\core\Dispatcher::invokeMethod(Array, Array)
#16 C:\wamp\www\timeclock\index.php(8): Flight::__callStatic('start', Array)
#17 C:\wamp\www\timeclock\index.php(8): Flight::start()
#18 {main}

The issue seems to be on line #323 of Engine.php in the _stop function.
the passed value for $code is sometimes an empty array instead of an int.

Here's a ghetto patch, I don't know much about the internals of this framework.

``` php
public function _stop($code = 200) {
        if(is_array($code))
          $code = 200;
        $this->response()
            ->status($code)
            ->write(ob_get_clean())
            ->send();
    }
```

**Comment by mikecao** [comment]
Thanks for the report. Now fixed in the latest commit 001c8ec301ab85d74232d7b250808fcae0e19cad.

**Comment by echelon** [comment]
Mike, I just wanted to report that I ran into this problem too. After your patch / latest commit, I now have a new problem:

```
Fatal error: Using $this when not in object context in 
[composer dir] /mikecao/flight/flight/Engine.php on line 296 
```

I'm working right now, but I might be able to get a better bug report soon. For now I'm going to roll with 5414b906f6c661801248345aba6d38d5e065295b

Thanks!

**Comment by mikecao** [comment]
@echelon I forgot that `$this` was not supported in closures in earlier versions of PHP. Now fixed in 86d1ea20e4a367093a723be086a7e5cfdbb9d406.

**Comment by Skinner927** [comment]
Thanks Mike, All seems well now.
